bundles/org.openhab.binding.keba/src/main/java/org/openhab/binding/keba/internal/handler/KeContactTransceiver.java

```diff
@@ -219,6 +219,9 @@ protected ByteBuffer send(String message, KeContactHandler handler) {
         while (true) {
             try {
                 synchronized (selector) {
+                    if (!selector.isOpen()) {
+                        return;
+                    }
```

Comment:
As far as I can tell `ClosedSelectorException` could be thrown from any of the selector operations and since the selector is closed from another thread it could be closed on this thread at any time. I don't think that simply checking if it is open at this point is sufficient to prevent `ClosedSelectorException` from getting thrown.

I think the easiest way to deal with this error is to just catch the `ClosedSelectorException`. 

But a more proper way to prevent this error from getting thrown is to make the transceiverThread responsible for opening and closing the selector and to add logic to allow outside threads to instruct the transceiverThread to close the selector on its own accord.

I'm ok with either solution but I think the first one is better because there are fewer things that can go wrong with it.

Reply:
Thanks @cpmeister, this makes a lot of sense indeed. I have updated the PR accordingly.